WordPress/gutenberg
# Data views: Add ability to display fields as a badge in grid layout

## What?
This PR does a few things which can be boiled down to three main changes:

1. Allow data view consumers to specify that certain fields should appear as a label-less 'badge' in grid layout.
2. Display 'badge' fields together above all others.
3. Update the Patterns page so that the 'sync status' field displays as a badge, and style accordingly.

## Why?
Follow-up to https://github.com/WordPress/gutenberg/pull/60083, which introduced the ability to arrange fields in a 'column' layout and forced the display of field labels.

This change is for fields where the label is not necessary, IE it can be interpreted from the value alone. 

Of the current field set across all data views, 'sync status' in patterns is an appropriate consumer: There is no need for the verbose: "Sync status: Synced" when "Synced" alone will do.

As data views expand to other areas (e.g. Media Library: https://github.com/WordPress/gutenberg/issues/55238) this affordance will become more useful. Media is a good example because media files have many fields that do not require a label (size, file name, type, dimensions, etc.). This PR will make it possible to implement the following mockup:

![Export](https://github.com/WordPress/gutenberg/assets/846565/d075edbf-6753-48cc-8738-3dfbcec1dc39)

## How?
1. Adds `displayAsBadgeFields` parameter to `GridItem`, similar to `displayAsColumnFields`
2. Detect and display all 'badge fields' in a new `HStack` _above_ all other fields.
3. Include `field.id` as a classname on all field outputs.
4. Add basic styling for badges.
5. Update `defaultConfigPerViewType` in the Patterns page to specific that `sync-status` is a badge field.
6. Update `sync-status` field render to include class name based on value
7. Apply purple accents to synced pattern badges

Two other minor tweaks:

1. Sync status field name: "Sync Status" -> "Sync status".
2. In pages data view, "Date" field name is now more explicit: "Publish date".

## Testing Instructions
1. Open the patterns page.
2. Toggle on the 'Sync status' field.
3. Observe the new display.
4. Check that all other data views and layouts work as before.

### Before

<img width="982" alt="Screenshot 2024-03-28 at 12 53 00" src="https://github.com/WordPress/gutenberg/assets/846565/cb8f50fe-3cd5-42ea-95d3-92a46bb20379">

### After

<img width="979" alt="Screenshot 2024-03-28 at 12 53 10" src="https://github.com/WordPress/gutenberg/assets/846565/ae8b5137-50c1-4fd0-8d8d-f6b19a089d32">


For more advanced testing you can temporarily swap fields in the Pages data view to display as badges by opening `packages/edit-site/src/components/sidebar-dataviews/default-views.js` and updating `DEFAULT_CONFIG_PER_VIEW_TYPE` like so:

```
export const DEFAULT_CONFIG_PER_VIEW_TYPE = {
	[ LAYOUT_TABLE ]: {
		primaryField: 'title',
	},
	[ LAYOUT_GRID ]: {
		mediaField: 'featured-image',
		primaryField: 'title',
		displayAsBadgeFields: [ 'author', 'status' ],
	},
	[ LAYOUT_LIST ]: {
		primaryField: 'title',
		mediaField: 'featured-image',
	},
};
```

Then:

1. Open the Pages data view and switch to Grid layout.
2. Toggle the different fields.
3. Observe that Author and Status appear as badges like so:

https://github.com/WordPress/gutenberg/assets/846565/a6575e85-59f8-4a10-92de-8e5157297ded

Note: It may be desirable to display Status as a badge with different styling for different values, similar to sync status. We can explore that in a follow up.
